server/src/main/java/timely/Server.java

```diff
@@ -142,25 +139,15 @@ private static boolean useEpoll() {
         }
     }
 
-    private static String usage() {
-
-        return "Server <configFile>";
-    }
-
     public static void fatal(String msg, Throwable t) {
         LOG.error(msg, t);
         LATCH.countDown();
     }
 
     public static void main(String[] args) throws Exception {
 
-        if (args.length != 1) {
-            System.err.println(usage());
-        }
-        final File conf = new File(args[0]);
-        if (!conf.canRead()) {
-            throw new RuntimeException("Configuration file does not exist or cannot be read");
-        }
+        Configuration conf = initializeConfiguration(args);
+
         Server s = new Server(conf);
```

Comment:
this might not be the right place to comment on this, but I dislike how creating a server object has side effects. I would much rather have the Server implement a Server interface and have methods like configure, start, stop. Then we could have a single Main that parses args, injects the proper server (real or standalone), etc. I'm bringing it up here because there might be a Spring way of doing this. I've only done injection like this with Guice.